### PR TITLE
fix 'play from here' context menu item

### LINKED ISSUE #18347: Kodi v18.8 on Nvidia Shield missing Play Only, when autoplaynext is enabled
### Describe the bug
when you have any combination of Auto Play Next for Video Items configured, it will only show "Play from here" as an option, and the "Play only this" is never shown.  As the code still has the "Play only this" context menu still in the code, but there was a change  to the videoplayer.autoplaynext option from a boolean to a string, it seems that this code was missed in xbmc/video/windows/GUIWindowVideoBase.cpp on line 892.

## Expected Behavior
When auto play is enabled, the context menu "Play from here", should change to "Play only this"

## Actual Behavior
No matter the the configuration only "Play from here" is in the context menu.

### To Reproduce
1. Turn on Autoplaynext for the video 
2. go to a movie and open up the context menu

### Your Environment
Used Operating system:

 - [x ] Android
 - [ ] iOS
 - [ ] Linux
 - [ ] OSX
 - [ ] Raspberry-Pi
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name:
 - Kodi version:

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

### PR DESCRIPTION
**Pull request by ronie**
## Description
in https://github.com/xbmc/xbmc/pull/13992 the `videoplayer.autoplaynext` setting has been changed from a bool to a list of int's.

the code in GUIWindowVideoBase.cpp that toggles one of the context menu items from '_Play from here_' to '_Play only this_' based on that setting, was not updated to adopt this.

as a result it will currently always add '_Play from here_'  to the context menu.

## Motivation and Context
fixes https://github.com/xbmc/xbmc/issues/18347

## How Has This Been Tested?
tested in Estuary, using various combinations of the `videoplayer.autoplaynext` setting.
checked the context menu on a movie, episode, musicvideo and an unclassified video
and confirmed it correctly shows either '_Play from here_' or '_Play only this_' based on the value of `videoplayer.autoplaynext`

## Types of change

- [X] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/video/windows/GUIWindowVideoBase.cpp [hunk 1]
```diff
@@ -73,6 +74,11 @@ using namespace KODI::MESSAGING;
 #define PROPERTY_GROUP_BY           "group.by"
 #define PROPERTY_GROUP_MIXED        "group.mixed"
 
+static const int SETTING_AUTOPLAYNEXT_MUSICVIDEOS = 0;
+static const int SETTING_AUTOPLAYNEXT_EPISODES = 2;
+static const int SETTING_AUTOPLAYNEXT_MOVIES = 3;
+static const int SETTING_AUTOPLAYNEXT_UNCATEGORIZED = 4;
```
Comment (ksooo) [inline xbmc/video/windows/GUIWindowVideoBase.cpp:80]: static constexpr int ..., please

File: xbmc/video/windows/GUIWindowVideoBase.cpp [hunk 2]
```diff
@@ -915,10 +921,28 @@ void CGUIWindowVideoBase::GetContextButtons(int itemNumber, CContextButtons &but
...
           m_vecItems->Size() > 1 && itemNumber < m_vecItems->Size() - 1)
       {
-        if (!CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(CSettings::SETTING_VIDEOPLAYER_AUTOPLAYNEXTITEM))
-          buttons.Add(CONTEXT_BUTTON_PLAY_AND_QUEUE, 13412);
-        else
+        int settingValue = SETTING_AUTOPLAYNEXT_UNCATEGORIZED;
+
+        if (item->IsVideoDb() && item->HasVideoInfoTag())
+        {
+          std::string mediaType = item->GetVideoInfoTag()->m_type;
```
Comment (ksooo) [inline xbmc/video/windows/GUIWindowVideoBase.cpp:928]: const std::string

File: xbmc/video/windows/GUIWindowVideoBase.cpp [hunk 3]
```diff
@@ -915,10 +921,28 @@ void CGUIWindowVideoBase::GetContextButtons(int itemNumber, CContextButtons &but
...
+            settingValue = SETTING_AUTOPLAYNEXT_EPISODES;
+          else if (mediaType == MediaTypeMovie)
+            settingValue = SETTING_AUTOPLAYNEXT_MOVIES;
+        }
+
+        const auto setting = std::dynamic_pointer_cast<CSettingList>(
+                   CServiceBroker::GetSettingsComponent()->GetSettings()->GetSetting(
+                   CSettings::SETTING_VIDEOPLAYER_AUTOPLAYNEXTITEM));
+
+        if (setting != nullptr && CSettingUtils::FindIntInList(setting, settingValue))
```
Comment (ksooo) [inline xbmc/video/windows/GUIWindowVideoBase.cpp:942]: if (setting && CsettingsUtils...

File: xbmc/video/windows/GUIWindowVideoBase.cpp [hunk 4]
```diff
@@ -915,10 +921,28 @@ void CGUIWindowVideoBase::GetContextButtons(int itemNumber, CContextButtons &but
...
+
+          if (mediaType == MediaTypeMusicVideo)
+            settingValue = SETTING_AUTOPLAYNEXT_MUSICVIDEOS;
+          else if (mediaType == MediaTypeEpisode)
+            settingValue = SETTING_AUTOPLAYNEXT_EPISODES;
+          else if (mediaType == MediaTypeMovie)
+            settingValue = SETTING_AUTOPLAYNEXT_MOVIES;
+        }
+
+        const auto setting = std::dynamic_pointer_cast<CSettingList>(
```
Comment (ksooo) [inline xbmc/video/windows/GUIWindowVideoBase.cpp:938]: I would not use auto here as type CSettingsList* is not really complex. IMO, explicitly naming the type improves readability of the code.

Comment (ronie) [reply xbmc/video/windows/GUIWindowVideoBase.cpp:938]: @ksooo thanx for the review :-)
fixed the three issues above.

this one was a c&p from Montellese's code, assuming he knows best when it comes to settings stuff:
https://github.com/xbmc/xbmc/blob/4c9f798b44c87b8197c3632f758963728cabb367/xbmc/view/GUIViewState.cpp#L514-L516

**Comment by ronie** [comment]
jenkins build this please

**Review by ksooo — APPROVED** [review]
lgtm

**Pull request merged**